sentinel-ai-models-simple-openai/src/main/java/com/phonepe/sentinelai/models/SimpleOpenAIModel.java

```diff
@@ -1118,4 +1123,43 @@ private ToolChoiceOption computeToolChoice(OutputGenerationMode outputGeneration
         };
     }
 
+    private void executeMessagesPreProcessors(final List<AgentMessagesPreProcessor> messagesPreProcessors,
+                                              final ModelUsageStats statsForRun,
+                                              final ModelRunContext context,
+                                              final ArrayList<AgentMessage> allMessages,
+                                              final ArrayList<ChatMessage> openAiMessages,
+                                              final ArrayList<AgentMessage> newMessages) {
+        List<AgentMessage> transformedMessages = allMessages;
```

Comment:
var

Reply:
`transformedMessages = Objects.requireNonNullElseGet(allMessages, List::<AgentMessage>of)`